playground/backend/internal/fs_tool/fs.go

```diff
@@ -0,0 +1,57 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fs_tool
+
+import (
+	pb "beam.apache.org/playground/backend/internal/api"
+	"fmt"
+	"github.com/google/uuid"
+)
+
+// FileSystemService interface for all type(Java/Go/Python/Scio) of FileSystem services.
+type FileSystemService interface {
```

Comment:
May we consider breaking up this interface to follow more of the typical interface to method 1:1 carnality we see with go?  Brainstorming out loud, what do we think about the following?  The following isn't complete but just a conversation starter.  I removed the `bool` from the return signatures as it could be part of the error.

```
import "io/fs"

type Setup interface {
    Create() (fs.FS, error)
}

type Teardown interface {
    Delete() error
}

type LifeCycle struct {
    sdk pb.Sdk
    root fs.FS
    folderGlobs []string
}

func New(ctx context.Context, sdk pb.Sdk) (*LifeCycle, error) {
        switch sdk {
	case pb.Sdk_SDK_JAVA:
		return &LifeCycle{
                     sdk: sdk,
                     root: javaFS, // See java_fs.go
                     folderGlobs: javaGlobs,
                }, nil
	default:
		return nil, fmt.Errorf("%s isn't supported now", sdk)
	}
}

func (lc *LifeCycle) Create() (fs.FS, error) {
    for _, k := range lc.folderGlobs {
        if err := MkdirAll( ... ); err != nil {
             ...
        }
    }
}

func (lc *LifeCycle) Delete() error {
    ....
}
```